problem_set_1.py

```diff
@@ -89,5 +156,57 @@
 # g = a
 # output g * 2**d
 
+
 # Write a function called `gcd` that takes two arguments and returns the greatest
 # common divisor using the instructions above.
+def gcd(a, b):
+    # TRYING TERNARY OPERATORS--THIS DIDN'T WORK
+    # d = 0
+    # while is_even(a) and is_even(b):
+    #     a = a / 2
+    #     b = b / 2
+    #     d = d + 1
+    # while a != b:
+    #     a = a / 2 if is_even(a) else b = b / 2 if is_even(b) else a = (a - b) / 2 if a > b else b = (b - a) / 2
+    # ALTERNATELY===
+    # a = a / 2 if is_even(a) else (a - b) / 2
+    # b = b / 2 if is_even(b) else (b - a) / 2
+    # ===
+    # g = a
+    # return g * 2**d
+
+    # TRYING A TUPLE WITH TERNARY OPERATORS
+    counter = 0
+    while is_even(a) and is_even(b):
+        a = a / 2
+        b = b / 2
+        counter += 1
+    ab_tuple = (a, b)
+    while a != b:
+        while is_even(a) or is_even(b):
+            ab_tuple = (a/2, b/2) if is_even(a) and is_even(b) else (a/2, b) if is_even(a) \
+            and not is_even(b) else (a, b/2) if not is_even(a) and is_even(b) else (a, b)
+            a = ab_tuple[0]
+            b = ab_tuple[1]
+        ab_tuple = ((a-b)/2, b) if a > b else (a, (b-a)/2)
+        a = ab_tuple[0]
+        b = ab_tuple[1]
+    return a * 2**counter
```

Comment:
@rlconley SO I feel like I absolutely brute-force shoehorned both ternary operators AND a tuple in this alternate solution in a way god surely never intended, but BY GOLLY it passes the tests!